### ISSUE TITLE
hotInstance.getCell not working in Jest

### ISSUE THREAD

**Issue by mikeyrt16** [handsontable/handsontable#8472]
### Description
When trying to test a table in Jest, hotInstance.getCell() returns null for all table columns except the first one. I have custom renderers I need to test that appear on the 2nd column and after, which are currently impossible to test due to this issue.

hotInstance.countVisibleCols() is returning 0, which feels like an insight into why it isn't working.

I've looked at your handsontable tests for countVisibleCols which set the table width to an appropriate value, which I've tried in the React wrapper to no avail.

### Steps to reproduce
```
it('should be able to get the cell from any column', async () => {
  const wrapper = mount(
    <Table
      columns={[
        { data: 'A', type: 'numeric' },
        { data: 'B', type: 'numeric' },
        { data: 'C', type: 'numeric' },
      ]}
      data={[
        [{ A: 123 }, { B: 456 }, { A: 789 }],
        [{ A: 123 }, { B: 456 }, { A: 789 }],
        [{ A: 123 }, { B: 456 }, { A: 789 }],
      ]}
    />,
    {
      attachTo: document.body.querySelector<HTMLElement>('#hotContainer'),
    },
  )
  await sleep(100)
  const { hotInstance } = wrapper.find(HotTable).instance() as HotTable

  expect(hotInstance.getCell(0, 0)).not.toBeNull() // Passes
  expect(hotInstance.getCell(0, 1)).not.toBeNull() // Fails...
  expect(hotInstance.getCell(0, 2)).not.toBeNull() // Fails...

  expect(hotInstance.getCell(1, 0)).not.toBeNull() // Passes
  expect(hotInstance.getCell(1, 1)).not.toBeNull() // Fails...
  expect(hotInstance.getCell(1, 2)).not.toBeNull() // Fails...

  expect(hotInstance.countVisibleCols()).toBe(0) // Passes????...

  wrapper.detach()
})
```

### Your environment
* React wrapper version: 8.4.0
* Handsontable version: 8.4.0
* Browser Name and version: N/A
* Operating System: macOS Catalina 10.15.5

**Comment by swistach** [comment]
Hi @mikeyrt16,
Our `getCell()` method bases on the DOM directly.
The virtual rows/column rendering uses `document.body` dimensions to set handsontable's overlays size. In jest, JSDom is used to simulate a "browser" environment - unfortunately, it's not simulating "real" elements' rendering. As a result, we get a `document.body` that dimensions are 0 (for height and width).

I found an issue on the `react-testing-library` repository where someone reported a similar problem. You can find it here: testing-library/react-testing-library#353.
Long story short - you can try workaround `jest` and `jsdom` configuration or change test runner to the other tool (puppeteer, karma, jasmine, cypress - anything that runs real browser - even in headless mode).

**Comment by mikeyrt16** [comment]
Thanks for the rely @swistach 

In a Jest context, how would you recommend I write a test where I need to interact with any cell after the first column?

For example - If I wanted to click a button in column x and assert against some side effect
![image](https://user-images.githubusercontent.com/10453522/127349444-0c49d599-6996-4e26-bc3b-b5be4b816008.png)

**Comment by swistach** [comment]
You can try to configure `puppeteer` within the `jest` environment - https://jestjs.io/docs/puppeteer.
Alternatively, you can use other tools to run integration tests (Cypress, Karma+Jasmine, etc.).

**Comment by mikeyrt16** [comment]
That's great, cheers. I appreciate you taking the time.
